be/src/storage/rowset/map_column_iterator.cpp

```diff
@@ -148,8 +177,17 @@ Status MapColumnIterator::next_batch(const SparseRange& range, Column* dst) {
 
     // if array column is nullable, element_read_range may be empty
     DCHECK(element_read_range.empty() || (element_read_range.begin() == _keys->get_current_ordinal()));
-    RETURN_IF_ERROR(_keys->next_batch(element_read_range, map_column->keys_column().get()));
-    RETURN_IF_ERROR(_values->next_batch(element_read_range, map_column->values_column().get()));
+    if (_access_keys) {
+        RETURN_IF_ERROR(_keys->next_batch(element_read_range, map_column->keys_column().get()));
```

Comment:
how about set _keys to default_value_iterator if access_keys is false

Reply:
I think OK, but default_value_iterator need some special config, I can test it later